vectordraw/vectordraw.py

```diff
@@ -0,0 +1,541 @@
+        'custom_checks'
+    )
+
+    has_score = True
+
+    @property
+    def settings(self):
+        """
+        Return settings for this exercise.
+        """
+        width_scale = self.width / float(self.height)
+        box_size = self.bounding_box_size
+        bounding_box = [-box_size*width_scale, box_size, box_size*width_scale, -box_size]
+        return {
+            'width': self.width,
+            'height': self.height,
+            'bounding_box': bounding_box,
+            'axis': self.axis,
+            'show_navigation': self.show_navigation,
+            'show_vector_properties': self.show_vector_properties,
+            'show_slope_for_lines': self.show_slope_for_lines,
+            'add_vector_label': self.add_vector_label,
+            'vector_properties_label': self.vector_properties_label,
+            'background': self.background,
+            'vectors': self.get_vectors,
+            'points': self.get_points,
+            'expected_result': self.get_expected_result,
+            'expected_result_positions': self.expected_result_positions,
+        }
+
+    @property
+    def user_state(self):
+        """
+        Return user state, which is a combination of most recent answer and result.
+        """
+        user_state = self.answer
+        if self.result:
+            user_state['result'] = self.result
+        return user_state
+
+    @property
+    def background(self):
+        """
+        Return information about background to draw for this exercise.
+        """
+        return {
+            'src': self.background_url,
+            'width': self.background_width,
+            'height': self.background_height,
+            'description': self.background_description,
+        }
+
+    def _get_default_vector(self):  # pylint: disable=no-self-use
+        """
+        Return dictionary that represents vector with default values filled in.
+        """
+        return {
+            'type': 'vector',
+            'render': False,
+            'length_factor': 1,
+            'length_units': '',
+            'base_angle': 0,
+            'style': {
+                'pointSize': 1,
+                'pointColor': 'red',
+                'width': 4,
+                'color': 'blue',
+                'label': None,
+                'labelColor': 'black'
+            }
+        }
+
+    @property
+    def get_vectors(self):
+        """
+        Return info about vectors belonging to this exercise.
+
+        To do this, load vector info from JSON string specified by course author,
+        and augment it with default values that are required for rendering vectors on the client.
+        """
+        vectors = []
+        for vector in json.loads(self.vectors):
+            default_vector = self._get_default_vector()
+            default_vector_style = default_vector['style']
+            if 'style' in vector:
+                default_vector_style.update(vector['style'])
+                del vector['style']
```

Comment:
For these cases where you have to use and remove a value from a dict, `d.pop(k)` is useful.